### ISSUE TITLE
[MICROGRANT] IPNS-Link

### ISSUE THREAD

**Issue by SomajitDey** [ipfs/devgrants#122]
### 1. What is your project? (max 100 words)

**Main Goal**: Addresses/Exposes any website, even a dynamic one, using IPNS and serve efficiently using IPFS.

**Overview**: Alice runs her http-server (website) on localhost, and also runs the [ipns-link](https://github.com/ipns-link/ipns-link/blob/main/tutorials/QuickStart.md) app on the same machine. The app gives her an IPNS-name. Even if her localhost doesn't have a public IP, her website becomes accessible via any gateway, using the IPNS-Name. IPFS is used as a CDN and for making live-streaming efficient.

**Catchphrase**: **Free, secure, censorship-resistant, domainless self-hosting for the decentralized web.**

GitHub organization: https://github.com/ipns-link
  

Website: https://www.ipns.live

### 2. How are you planning to improve this project? (max 200 words)

Priority wise:

- Scale up and maintain our existing public gateway and host more gateways distributed across the globe; i.e. **server costs**.
- Port the existing codebase to Go for speed and efficiency. Developer time needs to be incentivized.
- Implement a planned `.ipns` domain system for mapping long IPNS names to short human-friendly names. Centralized registrar with decentralized registry/index distributed using IPFS. Names to be given free of cost to students and others who can't afford a standard domain name.
- Implement a decentralized search engine for all websites exposed using IPNS-Link with a .ipns name.
- Host public relays to better serve users behind NAT.
- Port codebase to JS so private gateways can be run in the browser.
- Create a mobile-app backend service.

 ### 3. Will the work be Open Source?

Yes.

### 4. If selected, do you agree to complete weekly updates and a grant report upon conclusion?

Sure. We are very thorough about documenting everything as we progress.

### 5. Does your proposal comply with our Community Code of Conduct?

Yes.

### 6. Links and submissions

  
* Have you submitted to the IPFS Community Showcase ([instructions](https://github.com/ipfs/community/blob/master/README.md#showcase-your-project))?
  
Yes. Here it is: https://github.com/ipfs/community/pull/692. But it's been sitting there for 10 days, without any review or comments.

* Have you added your project to the [IPFS Ecosystem Directory](https://airtable.com/shrjwvk9pAeAk0Ci7)? 
 Yes
 
* Have you filled out the [Interplanetary Builder Feedback Survey](https://airtable.com/shrDZMizx03jOa4mQ)? 
 Yes
 
### Additional questions:
* For each team member(s), please list name, email, Github account, and role in the project.

1. Somajit Dey, somajit@ipns.live, @SomajitDey, Role: Maintenance and coding
2. Jonathan Miller, cloudy@ipns.live, @rinzler100, Role: Webmaster and webdev
3. LynHyper, @LynHyper, Role: Documentation, coordination and promotion

**Comment by SeedingTrees** [comment]
This microgrant has been approved! Look out for an email to submit payment details. Let's use your existing Community Discussion thread for updates.

**Comment by rinzler100** [comment]
Hello! I am a member of the IPNS link team. I am curious if you could validate in any way that fil.org is associated with ipfs devgrants. I couldn't find anything on the internet for this and submitting personal details to an organization I can't track the origin of is a bit disturbing...

**Comment by SeedingTrees** [comment]
Thanks, Rinzler.  Yes, that is the website for the Filecoin Foundation.  I am a Senior Fellow in the Foundation.  If your grant proposal is accepted, you may find yourself interacting with folks from either the Filecoin Foundation or Protocol Labs.

**Comment by SomajitDey** [comment]
# Announcement:
@SomajitDey is leaving the IPNS-Link project due to lack of time
@rinzler100 will be accepting the grant on behalf of the existing team

**Comment by rinzler100** [comment]
Hello, I ask this on behalf of the IPNS-Link team: is there an ETA on the delivery of the funds? (We know it will come in within a month but we were curious as to when so we can make better informed decisions)

**Comment by SeedingTrees** [comment]
Thanks for checking in, @rinzler100.  Did you receive the email that I sent on December 6 with the instructions for requesting payment?  If so, please reply to that email so that we can work through the payment details.

**Comment by rinzler100** [comment]
Alright, I sent the question via the second email thread (the latest one), thanks.

**Comment by SeedingTrees** [comment]
Thank you.  I am adding in an additional distro to help you with receiving
your microgrant.

Cheers,

Hunter

[image: Logo]

*Hunter Treseder*

Head of Social Impact

Filecoin Foundation & Filecoin Foundation for the Decentralized Web

✆: (415) 935-3249

Fil.org <https://fil.org> // FFDWeb.org <http://ffdweb.org>

**Comment by rinzler100** [comment]
Can you clarify what you mean by "additional distro"?

**Comment by SeedingTrees** [comment]
I did not realize that your message was from Github.  I was moving through emails quickly and replied to your message like it was an email, including adding in the email distro of my colleagues who handle payments.

Let's move this conversation to email rather than Github.  We handle all of the payment discussions over email to protect personal details.  Please reply to my December 6 email with confirmation that you submitted your payment request per the instructions.

**Comment by SeedingTrees** [comment]
Hi IPNS-Link team,

Thanks for reaching out. We are working to deliver payments as quickly as
possible. We are a small team cranking out hundreds of disbursements.

Unfortunately, I cannot tell you right now which batch your payment is in,
however, your claim has been reviewed and approved for payment delivery.

Please do let us know should you not receive it by 1/10/22.

Cheers,
Protocol Labs Grants team

**Comment by rinzler100** [comment]
Thanks a ton for the eta, totally understand about the delay and we of course are super thankful for what your team does and how you empower others to make a change, have a wonderful holiday :)

**Comment by rinzler100** [comment]
Hello, I was told to contact you if we should not receive the funds by 1/10/22… As you might have guessed we have not :/ 


Regards,
Rinzler100

**Comment by ErinOCon** [comment]
@rinzler100 I am very sorry for the delay in reply. Were you able to receive the grant successfully? Our records reflect that the funds were delivered.

**Comment by ErinOCon** [comment]
@rinzler100, just circling back to confirm receipt of funds! Thanks!

**Comment by rinzler100** [comment]
Yes, we received the compensation funds successfully! Thanks a ton for helping our team add to the IPFS ecosystem!

Regards,
IPNS-Link Team
